testing/coordinator.go

```diff
@@ -26,26 +26,142 @@ var (
 type Coordinator struct {
 	t *testing.T
 
-	Chains map[string]*TestChain
+	CurrentTime time.Time
```

Comment:
Looks like all chains will be moving synchronously in time. This is fine as a default, but how would I create tests where chainA is at a different time than chainB, i imagine we'll want to create these tests to make sure the protocol works in asynchronous environments

Reply:
I added a function to update only one chain. The send message execution will now only update its own clock